diff --git a/index.js b/index.js
index a9609436fc0a4d09eb5223d8ff73802593956899..db9e21c6c6089a0afc78969aa929e003a917e0d9 100644
--- a/index.js
+++ b/index.js
@@ -254,7 +254,7 @@ class Package {
    * @param {String} depVersion semver
    * @param {String} savePrefix npm_config_save_prefix
    */
-  updateLocalDependency(resolved, depVersion, savePrefix) {
+  updateLocalDependency(resolved, depVersion, savePrefix, keepWorkspaceSpec = true) {
     const depName = resolved.name;
 
     // first, try runtime dependencies
@@ -275,6 +275,8 @@ class Package {
       if (!resolved.workspaceAlias) {
         const workspacePrefix = resolved.workspaceSpec.match(/^(workspace:[*|~|^]?)/)[0];
         depCollection[depName] = `${workspacePrefix}${depVersion}`;
+      } else if (!keepWorkspaceSpec) {
+        depCollection[depName] = `${savePrefix}${depVersion}`;
       }
     } else if (resolved.registry || resolved.type === "directory") {
       // a version (1.2.3) OR range (^1.2.3) OR directory (file:../foo-pkg)
