diff --git a/index.js b/index.js
index a22558e5c0b2779ac875d14288f8d67c22889839..d8a1f1f8514d5850eae24c23a9145ea0ebf5e608 100644
--- a/index.js
+++ b/index.js
@@ -5,6 +5,7 @@ const { ValidationError } = require("@lerna/validation-error");
 const { CyclicPackageGraphNode } = require("./lib/cyclic-package-graph-node");
 const { PackageGraphNode } = require("./lib/package-graph-node");
 const { reportCycles } = require("./lib/report-cycles");
+const { Project } = require("@lerna/project");
 
 /** @typedef {import("./lib/package-graph-node").PackageGraphNode} PackageGraphNode */
 
@@ -22,8 +23,14 @@ class PackageGraph extends Map {
    * @param {boolean} [forceLocal] Force all local dependencies to be linked.
    */
   constructor(packages, graphType = "allDependencies", forceLocal) {
+  
+  
     super(packages.map((pkg) => [pkg.name, new PackageGraphNode(pkg)]));
 
+    const project = new Project()
+    const allPackages = project.getPackagesSync()
+    const allPackagesMap = new Map(allPackages.map((pkg) => [pkg.name, new PackageGraphNode(pkg)]));
+
     if (packages.length !== this.size) {
       // weed out the duplicates
       const seen = new Map();
@@ -77,7 +84,7 @@ class PackageGraph extends Map {
           if (spec === "*" || spec === "^" || spec === "~") {
             workspaceAlias = spec;
             const prefix = spec === "*" ? "" : spec;
-            const version = depNode.version;
+            const version = depNode ? depNode.version : allPackagesMap.get(depName).version;
             spec = `${prefix}${version}`;
           }
         }
